cargo-miri/bin.rs

```diff
@@ -625,15 +625,6 @@ fn phase_cargo_rustc(mut args: env::Args) {
     let target_crate = is_target_crate();
     let print = get_arg_flag_value("--print").is_some(); // whether this is cargo passing `--print` to get some infos
 
-    // cdylib is just skipped, we cannot interpret it and do not need it
-    // for the rest of the build either.
-    if get_arg_flag_value("--crate-type").as_deref() == Some("cdylib") {
-        if verbose {
-            eprint!("[cargo-miri rustc] (cdylib skipped)");
-        }
-        return;
-    }
-
```

Comment:
The #1675 regression test still passed (both before and after f82c018c14a2457184576e6f2ba5a33d251fc3bc) even I remove this, so either that test did not work or this PR fixes #1675 in another way as well.
I also added a test for `cdylib`-only crate (since that is the "skipped" case if this is not removed, but this is removed anyway, so that test may be redundant).

Reply:
Do you mean the test added in https://github.com/rust-lang/miri/pull/1708? I did verify that removing the `cdylib` thing here makes that test fail, but you might have to `rm target -rf` to make sure it doesn't use cached results.

Reply:
Yes. IIRC I ran `cargo clean` before running tests. Sounds like this PR really fixes #1675 in another way.

Reply:
Sorry for the #1567 typo...

Reply:
[CI run](https://github.com/rust-lang/miri/runs/1898418367) for 45bcbfe65f0b6b2055a0b8f7f0f0fc8c59a1c272 (8d48ad908485a04112f27d6a22c85eae3e8e4aa7...45bcbfe65f0b6b2055a0b8f7f0f0fc8c59a1c272 does not change that test) confirmed this PR really fixes #1567.

Reply:
> Sounds like this PR really fixes #1567 in another way.

That's great, this fix makes more sense anyway. :)